Makefile

```diff
@@ -64,9 +66,16 @@ all: bin/base64 lib/libbase64.o
 bin/base64: bin/base64.o lib/libbase64.o
 	$(CC) $(CFLAGS) -o $@ $^
 
-lib/libbase64.o: $(OBJS)
-	$(LD) -r -o $@ $^
-	$(OBJCOPY) --keep-global-symbols=lib/exports.txt $@
+lib/exports.build.txt: lib/exports.txt
+ifeq (i686-w64-mingw32, $(TARGET))
+	sed -e 's/^/_/' $< > $@
+else
+	cp -f $< $@
+endif
```

Comment:
Honestly I'm not a big fan of this compatibility block, though I understand why it's needed. In the longer term I think we should try to get rid of the exports file entirely and rely on compiler/linker features to mark functions as hidden or visible.

Reply:
You mean like annotating these symbols as `BASE64_SYMBOL_PRIVATE`?

Reply:
I don't like this either but if we're to keep the current semantic, it's needed.
It's not the same as `BASE64_SYMBOL_PRIVATE`:
`BASE64_SYMBOL_PRIVATE` only acts on the visibility attribute of `GLOBAL` symbols e.g. even if `base64_stream_encode_plain` is marked as hidden it's still linkable (if building a static library or bare object).
`--keep-global-symbols` acts on `LOCAL`/`GLOBAL` attribute. It effectively marks symbols that are not in the list as `LOCAL` & those are not linkable anymore (as if we only had 1 translation unit - simulated by `ld -r` - and the function was declared `static`).

I wouldn't mind if the makefile produced the same things as the CMake build does for a static library (just something like `ar -o libbase64.a *.o`, getting rid of the `ld -r` and `objcopy`).

Another option is not to support makefile builds for Windows at all.

Reply:
My idea for the future roadmap is basically to engineer the library such that all functions start with `akbase64_` (which is probably what I will rename this library to, see #103), *and* that all functions are declared `static`, except for those that are part of the public API. Those two measures should remove all potential naming conflicts in practice.

Until then, I suggest we keep going as usual, so using this Makefile hack/complication out of necessity.